src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,127 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final int GROW_FACTOR = 2;
+    private Node<K, V>[] table;
+    private int size = 0;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int keyHash = hash(key);
+        int bucketNumber = getIndex(keyHash);
+        if (size == 0 || table[bucketNumber] == null) {
+            ifFullResize(table);
+            table[bucketNumber] = new Node<>(keyHash, key, value, null);
+            size++;
+            return;
+        }
+        addToBucketNextNode(key, value, keyHash, bucketNumber);
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int position = getIndex(hash(key));
+        Node node = table[position];
+        if (node == null) {
+            return null;
+        }
+        if (node.key == null) {
+            return (V) node.value;
+        } else {
+            while (node != null) {
+                if (Objects.equals(key, node.key)) {
+                    return (V) node.value;
+                }
+                node = node.next;
+            }
+            return null;
+        }
+    }
+
+    private int getIndex(int keyHash) {
+        return keyHash % table.length;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K, V> {
+        private final int hash;
+
+        private K key;
+
+        private V value;
+
+        private Node<K, V> next;
+
+        public Node(int hash, K key, V value, Node<K, V> next) {
+            this.hash = hash;
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private int hash(K key) {
+        int hash = key == null ? 0 : key.hashCode();
+        return hash < 0 ? -hash : hash;
+    }
+
+    private void addNode(K key, V value, int keyHash) {
+
+    }
+
+    private void addToBucketNextNode(K key, V value, int keyHash, int bucketNumber) {
```

Comment:
name doesn't really reflect what this method does, because it also has logic of replacing a value, and actually, it can be a part of the `put()` because it is it's only usage